src/test/java/ActionsTests.java

```diff
@@ -0,0 +1,84 @@
+//
+//        provideEmail("demo@class.com");
+//        providePassword("te$t$tudent");
+//        clickSubmit();
+//        choosePlaylistByName("Playlist Demo");
+//        displayAllSongs();
+//        Assert.assertTrue(getPlaylistDetails().contains(String.valueOf(countSongs())));
+//        //verifies if the playlist details song count is equal to the retrieved number of songs
+//    }
+//
+//    // Test #3 - Mouse Hover
+//    @Test
+//    public void hoverOverPlayButton()   {
+//        provideEmail("demo@class.com");
+//        providePassword("te$t$tudent");
+//        clickSubmit();
+//        Assert.assertTrue(hoverPlay().isDisplayed());
+//    }
+//    public WebElement hoverPlay()   {
+//        WebElement play = driver.findElement(By.cssSelector("[data-testid='play-btn']"));
+//        actions.moveToElement(play).perform();
+//        return wait.until(ExpectedConditions.visibilityOf(play));
+//    }
+//
+//    public void chooseAllSongsList() {
+//        wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("li a.songs"))).click();
+//    }
+//    public void contextClickFirstSong() throws InterruptedException {
+//        WebElement firstSongElement = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector(".all-songs tr.song-item:nth-child(1)")));
+//        actions.contextClick(firstSongElement).perform();
+//        Thread.sleep(2000);//added thread sleeps to show the right click during class (feel free to remove this)
+//    }
+//    public void choosePlayOption() {
+//        wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("li.playback"))).click();
+//    }
+//    public boolean isSongPlaying() {
+//        WebElement soundBarVisualizer = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[data-testid = 'sound-bar-play']")));
+//        return soundBarVisualizer.isDisplayed();
+//    }
+//    public void choosePlaylistByName(String playlistName) {
+//        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//a[contains(text(),'" + playlistName + "')]"))).click();
+//    }
+//    public int countSongs() {//counts songs inside playlist
+//        return driver.findElements(By.cssSelector("section#playlistWrapper td.title")).size();
+//    }
+//    public String getPlaylistDetails() { //retrieves playlist details from playlist header (displays number of songs in playlist)
+//        return driver.findElement(By.cssSelector("span.meta.text-secondary span.meta")).getText();
+//    }
+//    public void displayAllSongs() { //prints all songs inside the playlist - for demo purposes
+//        List<WebElement> songList = driver.findElements(By.cssSelector("section#playlistWrapper td.title"));
+//        System.out.println("Number of songs found: " + countSongs());
+//        for (WebElement e : songList) {
+//            System.out.println(e.getText());
+//        }
+//    }
+//}
```

Comment:
I recommend not using code comments: for, example, when you have a code that cannot be understood, then comments are not the solution. Try to rewrite the code or rename fields and other elements in a way that the reader will understand the action that you are doing there.